packages/sdk/src/marketplaces/approval.ts

```diff
@@ -0,0 +1,92 @@
+import { Signer } from "@ethersproject/abstract-signer";
+import { BaseProvider } from "@ethersproject/providers";
+import { NftSwapV4 } from "@traderxyz/nft-swap-sdk";
+
+import { Asset } from "../types";
+
+import type {
+  SwappableAssetV4,
+  UserFacingERC20AssetDataSerializedV4,
+  UserFacingERC721AssetDataSerializedV4,
+  UserFacingERC1155AssetDataSerializedV4,
+} from "@traderxyz/nft-swap-sdk";
+
+/**
+ * Approves asset for use on specified contract address.
+ * @param params Params for asset approval
+ */
+export async function approveAsset({
+  provider,
+  chainId,
+  walletAddress,
+  signer,
+  contractAddress,
+  asset,
+}: {
+  provider: BaseProvider;
+  chainId: number;
+  walletAddress: string;
+  signer: Signer;
+  contractAddress: string;
+  asset: Asset;
+}): Promise<void> {
+  const sdk = new NftSwapV4(provider, signer, chainId, {
+    zeroExExchangeProxyContractAddress: contractAddress,
+  });
```

Comment:
I cannot import `getApprovalStatus` and `approveAsset` directly from https://github.com/trader-xyz/nft-swap-sdk/blob/main/src/sdk/v4/pure.ts because webpack projects are unable to resolve it for some reason. Init'ing and using the sdk for its approval utilities is the next best option short of copy-pasting the codes from 0x v4 js